scripts/create-darwin-volume.sh

```diff
@@ -1,33 +1,235 @@
-#!/bin/sh
-set -e
+#!/usr/bin/env bash
+set -eu
 
-root_disk() {
-    diskutil info -plist /
-}
+# I'm a little agnostic on the choices, but supporting a wide
+# slate of uses for now, including:
+# - import-only: `. create-darwin-volume.sh no-main[ ...]`
+# - legacy: `./create-darwin-volume.sh` or `. create-darwin-volume.sh`
+#   (both will run main())
+# - external alt-routine: `./create-darwin-volume.sh no-main func[ ...]`
+if [ "${1-}" = "no-main" ]; then
+    shift
+    readonly _CREATE_VOLUME_NO_MAIN=1
+else
+    readonly _CREATE_VOLUME_NO_MAIN=0
+    # declare some things we expect to inherit from install-multi-user
+    # I don't love this (because it's a bit of a kludge)
```

Comment:
These are just covering enough surface area to avoid breaking existing workflows that directly invoke this script without args.

I don't personally think we should/need to support this long-term, but I think it should get top-billing/consideration in a separate changeset. If we decide to maintain that ability, I agree about re-cutting shared functionality.

Reply:
I've appended a longer dated comment here locally; sorry for not connecting the dots. :)

For reference:
```bash
    # CAUTION: (Dec 19 2020)
    # This is a stopgap. It doesn't cover the full slate of
    # identifiers we inherit--just those necessary to:
    # - avoid breaking direct invocations of this script (here/now)
    # - avoid hard-to-reverse structural changes before the call to rm
    #   single-user support is verified
    #
    # In the near-mid term, I (personally) think we should:
    # - decide to deprecate the direct call and add a notice
    # - fold all of this into install-darwin-multi-user.sh
    # - intentionally remove the old direct-invocation form (kill the
    #   routine, replace this script w/ deprecation notice and a note
    #   on the remove-after date)
    #
```